lib/improver/utilities/spatial.py

```diff
@@ -49,54 +49,77 @@
 MAX_DISTANCE_IN_GRID_CELLS = 500
 
 
-def check_if_grid_is_equal_area(cube):
+def check_if_grid_is_equal_area(cube, require_equal_xy_spacing=True):
     """Identify whether the grid is an equal area grid.
     If not, raise an error.
     Args:
         cube (iris.cube.Cube):
-            Cube with coordinates that will be cMAhecked.
+            Cube with coordinates that will be checked.
+        require_equal_spacing (bool):
+            Flag to require the grid is equally spaced in the two spatial
+            dimensions (not strictly required for equal-area criterion)
+
+    Raises:
+        ValueError: If coordinate points are not equally spaced along either
+            axis (from calculate_grid_spacing)
+        ValueError: If point spacing is not equal for the two spatial axes
+    """
+    xdiff = calculate_grid_spacing(cube, axis='x')
+    ydiff = calculate_grid_spacing(cube, axis='y')
+    if require_equal_xy_spacing and not np.isclose(xdiff, ydiff):
+        raise ValueError(
+            "Grid does not have equal spacing in x and y dimensions")
+
+
+def calculate_grid_spacing(cube, axis='x'):
+    """
+    Returns the grid spacing of a given spatial axis
```

Comment:
I think this function should do only what the title and doc-string suggest. It is valid to have an even grid spacing on a lat-lon grid and we will probably want this functionality for global data.
To achieve this, the units of the returned float need to be either "degrees" or "metres" with a try block to attempt both.
The error message to trap lat-lon coordinates should be moved to the check_if_grid_is_equal_area function.